### PR TITLE
Create Longform theme article page and style hero section

### PR DESCRIPTION
**Pull request by gabrieltanchen**
#### Description of changes
resolves CIGIHub/cigi-tickets#267

Screen Shot 2020-12-15 at 2 11 25 PM

### CODE REVIEW

**Comment by johnbxu** [comment]
image

- Hero section has different theming on smaller screens - font size, image overlay, social icons less spaced out.

**Comment by johnbxu** [comment]
image

Block quote margins should be decreased

File: templates/themes/longform/article_page.html [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        </ul>
+      </div>
+    </div>
+  </section>
+  <section class="body">
+    {% for block in self.body %}
+      {% include_block block %}
+    {% endfor %}
+  </section>
+{% endblock %}
```
Comment (johnbxu) [inline templates/themes/longform/article_page.html:61]: Template should include opinion disclaimer and about the author sections.

File: articles/models.py [hunk 2]
```diff
@@ -227,6 +227,12 @@ class HeroTitlePlacements(models.TextChoices):
     # Reference field for the Drupal-Wagtail migrator. Can be removed after.
     drupal_node_id = models.IntegerField(blank=True, null=True)
 
+    def get_template(self, request, *args, **kwargs):
+        standard_template = super(ArticlePage, self).get_template(request, *args, **kwargs)
+        if self.theme and self.theme.name == 'Longform':
```
Comment (johnbxu) [inline articles/models.py:232]: Not urgent, but this should probably automatically grab the corresponding template path based on theme name so it doesn't have a whole bunch of `if...else` s

**Comment by gabrieltanchen** [comment]
@johnbxu Updated - good catch on the mobile hero.

Updated the `ThemeablePageAbstract` model to dynamically get a template theme directory.

Updated the block quote margins in the general block quote template - the margins were a bit too large across the board. The Longform theme margin is still a bit bigger than what's on production, but I think it's better spaced out.

Also added the disclaimer and author sections. There's another issue to style the author section - this issue was just focused on the hero.

**Comment by johnbxu** [comment]
Somehow the subtitle is now displaying a `p` tag: 
image

File: templates/streams/block_quote_block.html [hunk 3]
```diff
@@ -1,34 +1,32 @@
 {% load wagtailcore_tags wagtailimages_tags %}
 
-<div class="stream-block stream-block-blockquote">
-  <div class="container-sm flex-row">
```
Comment (johnbxu) [inline templates/streams/block_quote_block.html:4]: Removing the `flex-row` makes the image stack above the rest: 
image

Also, should I change the other stream-blocks so the container class is on the same element as the `stream-block-x`?

**Comment by gabrieltanchen** [comment]
@johnbxu This is updated and ready for another review.

I updated the block quote to use flex and use the medium container if there's an image. There's only one node in Drupal that uses the block quote images and it uses the medium container: https://www.cigionline.org/articles/can-g20-save-globalizations-waning-reputation
Screen Shot 2020-12-17 at 11 08 27 AM

The `<p>` elements in the subtitle is a trickier one. That must have been because you saved the page. In Drupal this data is entered just as a character input, but in Wagtail I implemented it as a `RichTextField`, so that editors can add things like italics, etc. So the data in Wagtail _should_ have the `<p>` tag surrounding the text, which is what got updated when the page is saved. I updated the CSS to assume that the `<p>` tag is there. I'll have to update the migrator to insert the surrounding `<p>` tags when migrating from a plain text field.
Screen Shot 2020-12-17 at 11 08 03 AM

**Review by johnbxu — APPROVED** [review]

**Pull request merged**